### PR TITLE
Added quantity support for the !price command

### PR DESCRIPTION
**Pull request by johnhamilto**
Example:

!Price ranarr seed -> gives the same result as before these changes
!Price 67 ranarr seed -> gives the price of 67 ranarr seeds, as opposed to nothing at all before these changes

image

### CODE REVIEW

**Comment by Nightfirecat** [comment]
As was mentioned in Discord, this could fail on looking up items which begin with digits. (3rd age items might be the only ones) I'd strongly recommend writing some tests to cover cases like that.

**Comment by johnhamilto** [comment]
@Nightfirecat the plugin didn't match 3rd age items before, but now it uses regex instead of substrings so its more stable

File: runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java [hunk 1]
```diff
@@ -200,12 +200,23 @@ else if (config.lvl() && message.toLowerCase().equals(CMB_COMMAND_STRING))
 			log.debug("Running combat level lookup");
 			executor.submit(() -> combatLevelLookup(setMessage.getType(), setMessage));
 		}
+		else if (config.price() && message.toLowerCase().startsWith(PRICE_COMMAND_STRING + " ") && message.toLowerCase().matches("^![a-z]+ \\d+ .+"))
+		{
+			Pattern quantityPattern = Pattern.compile("^\\d+");
+			Pattern itemPattern = Pattern.compile("(?<=\\d ).+");
+			String input = message.substring(PRICE_COMMAND_STRING.length() + 1);
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java]: this defeats the whole purpose of using regex

File: runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java [hunk 2]
```diff
@@ -200,12 +200,23 @@ else if (config.lvl() && message.toLowerCase().equals(CMB_COMMAND_STRING))
 			log.debug("Running combat level lookup");
 			executor.submit(() -> combatLevelLookup(setMessage.getType(), setMessage));
 		}
+		else if (config.price() && message.toLowerCase().startsWith(PRICE_COMMAND_STRING + " ") && message.toLowerCase().matches("^![a-z]+ \\d+ .+"))
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java]: just use one regex statement matcher

File: runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java [hunk 3]
```diff
@@ -200,12 +200,23 @@ else if (config.lvl() && message.toLowerCase().equals(CMB_COMMAND_STRING))
...
+		{
+			Pattern quantityPattern = Pattern.compile("^\\d+");
+			Pattern itemPattern = Pattern.compile("(?<=\\d ).+");
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java]: precompile your regex at the start of the class

File: runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java [hunk 4]
```diff
@@ -202,10 +203,21 @@ else if (config.lvl() && message.toLowerCase().equals(CMB_COMMAND_STRING))
 		}
 		else if (config.price() && message.toLowerCase().startsWith(PRICE_COMMAND_STRING + " "))
 		{
-			String search = message.substring(PRICE_COMMAND_STRING.length() + 1);
-
-			log.debug("Running price lookup for {}", search);
-			executor.submit(() -> itemPriceLookup(setMessage.getMessageNode(), search));
+			Matcher priceMatcher = PRICE_PATERN.matcher(message.toLowerCase());
+			if (priceMatcher.matches())
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java]: this can be merged into into the first if check

File: runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java [hunk 5]
```diff
@@ -202,10 +203,21 @@ else if (config.lvl() && message.toLowerCase().equals(CMB_COMMAND_STRING))
 		}
 		else if (config.price() && message.toLowerCase().startsWith(PRICE_COMMAND_STRING + " "))
 		{
-			String search = message.substring(PRICE_COMMAND_STRING.length() + 1);
-
-			log.debug("Running price lookup for {}", search);
-			executor.submit(() -> itemPriceLookup(setMessage.getMessageNode(), search));
+			Matcher priceMatcher = PRICE_PATERN.matcher(message.toLowerCase());
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java]: We already know the string begins with `!price` here so you can just match `search` against `(?:(\\d*) )?(.+)`

File: runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java [hunk 6]
```diff
@@ -202,10 +203,21 @@ else if (config.lvl() && message.toLowerCase().equals(CMB_COMMAND_STRING))
...
+			if (priceMatcher.matches())
+			{
+				if (priceMatcher.group(1) != null)
+				{
+					System.out.println(priceMatcher.group(1));
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java]: Remove this

File: runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java [hunk 7]
```diff
@@ -202,10 +203,21 @@ else if (config.lvl() && message.toLowerCase().equals(CMB_COMMAND_STRING))
...
+					log.debug("Running price lookup for {}", priceMatcher.group(2));
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java]: Prefer using variables to hold variable state instead of duplicating code

File: runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java [hunk 8]
```diff
@@ -482,24 +495,41 @@ private void itemPriceLookup(MessageNode messageNode, String search)
 			int itemPrice = itemManager.getItemPrice(itemId);
 
 			final ChatMessageBuilder builder = new ChatMessageBuilder()
-				.append(ChatColorType.NORMAL)
+				.append(ChatColorType.HIGHLIGHT)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java]: Why was this changed? This probably should not be a part of this PR.

**Comment by Nightfirecat** [comment]
This looks like it uses the same format of chat as the item examining plugin--perhaps there's some shared code between them that could be reused?
